pkg/apis/v1alpha1/awsnodetemplate.go

```diff
@@ -104,3 +111,12 @@ type AWSNodeTemplateList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []AWSNodeTemplate `json:"items"`
 }
+
+// Placement contains the configuration for LaunchTemplates to run instances on dedicated hosts
+// https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata.html#cfn-ec2-launchtemplate-launchtemplatedata-placement
+type Placement struct {
+	// +optional
+	HostResourceGroupARN *string `json:"hostResourceGroupARN,omitempty"`
```

Comment:
Discussed this PR with the team today. Currently, we don't have any other api fields in Karpenter's CRDs that reference an ARN.  Could we make the HostResourceGroup referenced by name and then construct the arn for input within Karpenter? 

Same question for the License Configuration. It seems a little different since the name is generated. It may be a good place the use of a `selector` similar to the `subnetSelector` and `securityGroupSelector`.

Reply:
BTW, it does feel a bit odd writing selectors for things that aren't collections (in the Kubernetes sense).

Sometimes, when what you want is an ARN, and the API is AWS-specific, an ARN is going to be the right fit.

Reply:
I'll do some research into the AWS APIs but I don't believe these resources have any searchable properties. For example LicenseConfigurations are just ARNs - https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_LicenseConfiguration.html they don't have anything look-up-able.

Reply:
yeah I think this is the right conclusion. I would agree with your direction if there was some ability to query and discover license configurations, and host resource groups, but the APIs do not have anything I could find. Happy to be corrected if someone can provide the reference docs.

The reason we want ARNs and not just name's like "If you provide the license id 12345, karpenter will rewrite this as an arn", but this won't work in a number of scenarios:
* the resource is shared between AWS accounts (RAM)
* can we even find the account ID? what about aws partition for aws-gov environments?
Providing the full ARN bypasses all of these issues, and I think is the simplest and safest way to provide this feature.

Reply:
Synced in slack, @preflightsiren is kind enough to come to WG this Thursday at 2PM PDT to present this doc. Let's review there and sync up on this decision.

Reply:
I'm picking this up again this week, just refamiliarising myself and will push the new API up, perhaps as a new PR